src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,29 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """
+    Add target information to run_info under target_data
```

Comment:
Oh, also: can you mark this method ":API: public" to signal that is a pants public API? See http://www.pantsbuild.org/deprecation_policy.html